pkg/controller/sparkapplication/controller.go

```diff
@@ -566,7 +568,17 @@ func (c *Controller) syncSparkApplication(key string) error {
 			appCopy.Status.AppState.State = v1beta2.FailedState
 			appCopy.Status.AppState.ErrorMessage = err.Error()
 		} else {
-			appCopy = c.submitSparkApplication(appCopy)
+			// related to ofas, we received 2 times the CRD events with NewState state
+			// we need to check if the submission was already done
+			// and skip the submission if it was already done
+			if c.submissionCache.Exist(key) == false {
+				appCopy = c.submitSparkApplication(appCopy)
+				// whatever the result of the submission, we don't want to retry the submission
+				c.submissionCache.Set(key)
+			} else {
+				glog.V(2).Infof("submission attempt already done for: %q, skip it", key)
+				return nil
```

Comment:
maybe we shouldn't return here? in case there is other code further down in the method that still needs to run?

(now or in the future after pulling in from upstream)

Reply:
good question, hmmm
the second event in this case is not expected by the spark-operator,
I would say naively that  we should return asap and skip futur evolve from upstream

Reply:
I think it's ok. We respect the "original" behavior, i.e. `NewState` `?` -> `appCopy = c.submitSparkApplication(appCopy)` and then at the end we go here 

```go
if appCopy != nil {
		err = c.updateStatusAndExportMetrics(app, appCopy)
```

but if we receive the "duplicated" new status and we know that it is because of us and our watcher service, we'll ignore it.